src/runtime/test_case_runner.js

```diff
@@ -182,6 +200,25 @@ export default class TestCaseRunner {
     })
   }
 
+  async runStepHooks(step, hookDefinitions) {
+    const stepHookResults = await Promise.all(
+      hookDefinitions.map(async hookDefinition =>
+        this.runStepHook(step, hookDefinition)
+      )
+    )
+    const errors = stepHookResults.filter(result => result.exception)
+    if (errors && errors.length) {
+      return {
+        exception: 'Errors running step hooks',
+        status: Status.FAILED,
+      }
+    } else {
+      return {
+        status: Status.PASSED,
+      }
+    }
```

Comment:
I disagree with how this function works. I think we should run the step hooks sequentially and if any does not pass we return the step result of the not passing hook. If all pass, we return status passed.